app/bundles/LeadBundle/Entity/LeadListRepository.php

```diff
@@ -209,10 +209,12 @@ public function checkLeadSegmentsByIds(Lead $lead, $ids)
             ->where(
                 $q->expr()->andX(
                     $q->expr()->in('x.leadlist_id', $ids),
-                    $q->expr()->eq('l.id', ':leadId')
+                    $q->expr()->eq('l.id', ':leadId'),
+                    $q->expr()->in('x.manually_removed', ':manuallyRemoved')
```

Comment:
Why `in` and not
```suggestion
                    $q->expr()->eq('x.manually_removed', ':manuallyRemoved')
```
?

Reply:
Could you please also write test for this? I know that creating all the necessary campaign entities is time consuming. But creating a contact, 2 segments, add the contact to both, remove manually from one and then assert for the combination of the segments. An example to illustrate the idea:
```php
$this->em->getRepository(LeadList::class)->checkLeadSegmentsByIds(Contact, [$segmentA->getId(), $segmentB->getId()]);
```
I also noticed that the query can be optimized/simplified. It does not need the `leads` table at all. So we can avoid the join and change the condition: 
```diff
- $q->expr()->eq('l.id', ':leadId'),
+ $q->expr()->eq('x.lead_id', ':leadId'),
```
Do you think you could do that in this PR too? If not I can do it in a separate PR.

Reply:
I think I can do that here. And I'll add test.